controllers/restore/deploy.go

```diff
@@ -41,14 +48,90 @@ func (r *PulpRestoreReconciler) restorePulpCR(ctx context.Context, pulpRestore *
 		}
 
 		json.Unmarshal([]byte(cmdOutput), &pulp.Spec)
+
+		// store the number of replicas so we can rescale with the same amount later
+		podReplicas = PodReplicas{
+			Api:     pulp.Spec.Api.Replicas,
+			Content: pulp.Spec.Content.Replicas,
+			Worker:  pulp.Spec.Web.Replicas,
+			Web:     pulp.Spec.Web.Replicas,
+		}
+
+		pulp.Spec.Api.Replicas = 0
+		pulp.Spec.Content.Replicas = 0
+		pulp.Spec.Worker.Replicas = 0
+		pulp.Spec.Web.Replicas = 0
 		if err = r.Create(ctx, &pulp); err != nil {
 			log.Error(err, "Error trying to restore "+pulpRestore.Spec.DeploymentName+" CR!")
 			r.updateStatus(ctx, pulpRestore, metav1.ConditionFalse, "RestoreComplete", "Failed to restore cr_object!", "FailedRestore"+pulpRestore.Spec.DeploymentName+"CR")
-			return err
+			return PodReplicas{}, err
 		}
 
 		log.Info(pulpRestore.Spec.DeploymentName + " CR restored!")
 	}
 
+	return podReplicas, nil
+}
+
+// scaleDeployments will update pulp CR with 1 replica for each core component
+func (r *PulpRestoreReconciler) scaleDeployments(ctx context.Context, pulpRestore *repomanagerv1alpha1.PulpRestore, podReplicas PodReplicas) error {
+	log := r.RawLogger
+	pulp := &repomanagerv1alpha1.Pulp{}
+
+	if err := r.Get(ctx, types.NamespacedName{Name: pulpRestore.Spec.DeploymentName, Namespace: pulpRestore.Namespace}, pulp); err != nil && errors.IsNotFound(err) {
+		// Error reading the object - requeue the request.
+		log.Error(err, "Failed to get Pulp CR")
+		return err
+	}
+
+	/* Not sure if this is a good idea to promptly restore with the same number of replicas from backup.
+	   After a restore procedure it is common to manually double-check the integrity of the data and the health
+	   of the components. With fewer replicas it is easier to troubleshoot and identify possible reprovision issues.
+	   Another common scenario of the restore procedure is a DR. Some DR environments have less resources
+	   (because of cost, for example) than prod and starting with a single replica and letting users scale them
+	   manually also seems to be a good idea. */
```

Comment:
maybe we can add a variable to control it, a boolean so the user can opt if restore with 1 replica or with the same amount as before

Reply:
Good idea!
I'll open an issue for that.